src/Components/Navbar.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { getMissions } from '../redux/missions/missions';
+import icon from '../img/icon.png';
+import '../css/Navbar.css';
+
+export default function Navbar() {
+  const dispatch = useDispatch();
+  const stateM = useSelector((state) => state.missionsStore.missions);
+  const missionClickHandler = () => {
+    if (!stateM.length) {
+      dispatch(getMissions());
+    }
+  };
```

Comment:
- [ ] I think this piece of code in charge of fetching the data from the API and populating the missions state would be better placed in your missions container inside a `useEffect` hook, just as you did with the rockets one! 🤓 

There's a principle called "Single responsibility" in which a single function (in this case, component) should only be in charge of one responsibility, in this case, the Navbar has to deal with the navigation and also with fetching the mission's data, which doesn't make much sense! 😱 

Applying this change would give you a better separation of concerns, ending up with a Navbar component that's in charge only of the navigation, and a Mission component that's in charge of everything regarding missions! 🤓 🔧

Reply:
Yes, I was thinking to do that but the tasks for the project said that the fetch has to happen when the user navigates to the missions section.
![image](https://user-images.githubusercontent.com/114300121/216594510-afd727ba-0115-468a-b5fa-0ce3e38bca9c.png)
@anagudelogu

Reply:
Hi, @ShinhyoBelok thanks for reaching out! The key here is to understand what happens when the user `navigates` to the missions section:

This is the expected flow:

1. User clicks on the `missions` tab.
2. User should be redirected to `/missions`
3. The missions page renders the rocket data.

Let's notice the following: steps 1 and 2 are related to navigation only ⛵, we don't care about fetching the data at this stage, but the actual behavior is that when the user clicks on the missions button, the fetch action is getting fired (see first step) involving missions logic here.

Step 3 is the one related to rendering the missions page, in this step we do care about fetching the data because it is needed to render the actual component and display it to the end user, so using an `useEffect()` hook here will make sure that the data is fetched when the missions component is mounted.

So in reality you're still meeting the requirement, when the user navigates to the missions section, the missions page gets mounted and the data fetched. 🤓 

I hope this explanation let you understand better my suggestion, if not, feel free to reach out in slack, cheers, and happy coding! 🍾 🤓